packages/route/src/Infrastructure/SymfonyCmf/Routing/CmfRouteProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Route\Infrastructure\SymfonyCmf\Routing;
+
+use Sulu\Route\Application\Routing\Matcher\RouteCollectionForRequestLoaderInterface;
+use Symfony\Cmf\Component\Routing\RouteProviderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Route as SymfonyRoute;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @internal No BC promises are given for this class. It may be changed or removed at any time.
+ */
+final readonly class CmfRouteProvider implements RouteProviderInterface
+{
+    /**
+     * @param RouteCollectionForRequestLoaderInterface[] $routeCollectionForRequestLoaders
+     */
+    public function __construct(
+        private iterable $routeCollectionForRequestLoaders,
+    ) {
+    }
+
+    public function getRouteCollectionForRequest(Request $request): RouteCollection
+    {
+        foreach ($this->routeCollectionForRequestLoaders as $routeCollectionLoader) {
+            $routeCollection = $routeCollectionLoader->getRouteCollectionForRequest($request);
```

Comment:
I'm little bit unsure about giving here the Request `$request` here. I see that the Symfony router is also using the Request or string pathInfo for the UrlMatcher: https://github.com/symfony/routing/blob/ee9a67edc6baa33e5fae662f94f91fd262930996/Matcher/UrlMatcher.php#L85

But from my point of view the `RequestContext` would maybe be a better option 🤔  to be more consistent between `Matcher` and `Generator`, maybe I miss understand the concept of the RequestContext, and we should stay with Request for matching and context for generating.

/cc @Prokyonn @dbu

Reply:
as i understand, the matcher uses the Request as that holds a rich model of the whole request including headers. the header bag gives you precise access to headers, even parsing cache headers, or more relevant to routing the accept header etc.

Reply:
i would look if you can make this method a custom RouterInterface implementation that either throws the route not found exception or returns the one route that matches the slug.

the dynamic router is built around the idea of finding several candidates and not reimplement the logic to decide which route candidate should win, with accept or parameters and regex etc.

Reply:
Okay, the Symfony `RouterInterface` just has the `getRouteCollection` without any parameters and so `Request` so to achieve same logic I would create own router then and inject there the requestStack to get currentRequest for the matching. It doesn't make any problems implementing that interface and return a empty collection in none match / cli context? Thought maybe it would cache the empty collection and will then not longer be called again.

https://github.com/symfony/routing/blob/ee9a67edc6baa33e5fae662f94f91fd262930996/RouterInterface.php#L32